prism/src/main/java/org/apache/falcon/service/BacklogMetricEmitterService.java

```diff
@@ -0,0 +1,313 @@
+ */
+package org.apache.falcon.service;
+
+import org.apache.falcon.FalconException;
+import org.apache.falcon.LifeCycle;
+import org.apache.falcon.entity.EntityUtil;
+import org.apache.falcon.entity.v0.Entity;
+import org.apache.falcon.entity.v0.EntityType;
+import org.apache.falcon.entity.v0.SchemaHelper;
+import org.apache.falcon.entity.v0.process.Process;
+import org.apache.falcon.jdbc.BacklogMetricStore;
+import org.apache.falcon.metrics.MetricNotificationService;
+import org.apache.falcon.resource.InstancesResult;
+import org.apache.falcon.util.MetricInfo;
+import org.apache.falcon.util.StartupProperties;
+import org.apache.falcon.workflow.WorkflowExecutionContext;
+import org.apache.falcon.workflow.WorkflowExecutionListener;
+import org.apache.falcon.workflow.engine.AbstractWorkflowEngine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.falcon.workflow.WorkflowEngineFactory.getWorkflowEngine;
+
+/**
+ * Backlog Metric Emitter Service to publish metrics to Graphite.
+ */
+public final class BacklogMetricEmitterService implements FalconService,
+        EntitySLAListener, WorkflowExecutionListener {
+
+    private static final String METRIC_PREFIX = "falcon";
+    private static final String METRIC_SEPARATOR = ".";
+    private static final String BACKLOG_METRIC_EMIT_INTERVAL = "falcon.backlog.metricservice.emit.interval.millisecs";
+    private static final String BACKLOG_METRIC_RECHECK_INTERVAL = "falcon.backlog.metricservice."
+            + "recheck.interval.millisecs";
+
+    private static final Logger LOG = LoggerFactory.getLogger(BacklogMetricEmitterService.class);
+
+    private static BacklogMetricStore backlogMetricStore = new BacklogMetricStore();
+
+    private static final BacklogMetricEmitterService SERVICE = new BacklogMetricEmitterService();
+
+    private static MetricNotificationService metricNotificationService =
+            Services.get().getService(MetricNotificationService.SERVICE_NAME);
+
+    public static BacklogMetricEmitterService get() {
+        return SERVICE;
+    }
+
+    private BacklogMetricEmitterService() {
+    }
+
+    private ScheduledThreadPoolExecutor scheduledThreadPoolExecutor1 = new ScheduledThreadPoolExecutor(1);
+    private ScheduledThreadPoolExecutor scheduledThreadPoolExecutor2 = new ScheduledThreadPoolExecutor(1);
+
+
+    public static final ThreadLocal<SimpleDateFormat> DATE_FORMAT = new ThreadLocal<SimpleDateFormat>() {
```

Comment:
I am not using more features except parsing and formatting,  and also simple date format used lot of places in falcon.